realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -313,14 +309,19 @@ public void logout() {
                 }
             }
 
+            // Remove all local tokens, preventing further connections.
+            final Token userToken = syncUser.getUserToken();
+            syncUser.clearTokens();
+            syncUser.localLogout();
```

Comment:
Let's just leave them as two for now, but perhaps add a TODO about verifying or refactoring it?